### PR TITLE
[DPE-3974] Enable arm builds

### PR DESCRIPTION
**Pull request by dragomirp**
* Add arm builds
* Update dependencies
* Update charm lib

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,44 +39,43 @@ optional = true
...
-shellcheck-py = "^0.9.0.5"
+codespell = "^2.2.6"
+shellcheck-py = "^0.10.0.1"
 
 [tool.poetry.group.unit.dependencies]
-pytest = "^7.4.0"
-pytest-asyncio = "<0.23"
-coverage = {extras = ["toml"], version = "^7.4.1"}
+pytest = "^8.1.1"
+pytest-asyncio = "*"
```
Comment (dragomirp) [inline pyproject.toml:62]: This should only be needed to suppress warnings. We should just use whatever version `pytest-operator` is pulling

Comment (phvalguima) [reply pyproject.toml:62]: Agreed

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ async def test_build_and_deploy(ops_test: OpsTest, db_driver, use_router) -> Non
             apps=[DB_CHARM[db_driver]["app_name"]],
             status="active",
             raise_on_blocked=True,
-            timeout=15 * 60,
+            timeout=20 * 60,
```
Comment (phvalguima) [inline tests/integration/test_charm.py:147]: @dragomirp any particular reason to increase this value?

**Review by phvalguima — COMMENTED** [review]
@dragomirp the change mostly updates our CI. However, I'd like to see some test results, either you or someone manually executed them with sysbench; or we can extend one of our CIs (e.g. mysql / psql-operators) to use sysbench + ARM tests whenever they test.

File: tests/integration/test_charm.py (hunk 2 above)
Comment (taurus-forever) [reply tests/integration/test_charm.py:147]: Advanced sleep() programming! ;-)

**Review by taurus-forever — APPROVED** [review]
2 @phvalguima I do not see the huge CI update here (tnx for the shared workflow!).
LGTM for the changes. It looks like timeout increase based on the local tests. Up 2 Dragomir to reply.

File: tests/integration/test_charm.py (hunk 2 above)
Comment (dragomirp) [reply tests/integration/test_charm.py:147]: The one of the postgresql tests timed out [twice](https://github.com/canonical/sysbench-operator/actions/runs/8645051269/job/23702497568) in a row. The timeout used in postgresql VM tests is usually 25 mins.

**Comment by dragomirp** [comment]
> @dragomirp the change mostly updates our CI. However, I'd like to see some test results, either you or someone manually executed them with sysbench; or we can extend one of our CIs (e.g. mysql / psql-operators) to use sysbench + ARM tests whenever they test.

CI is AFAIK no go, due to cost of runners. We'll have to wait for self-hosted runners + support in data platform workflow for proper Arm CI. I can run some tests manually tomorrow though.

**Comment by taurus-forever** [comment]
re: Arm CI, just today we had the first Canonical self-hosted ARM runner to test.
As for now, run integration test for ARM only for this charm. Will be improved the nearest time!

**Review by phvalguima — APPROVED** [review]

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -11,26 +11,6 @@
...
-            # Building sysbench charm
-            return await _build_charm(
-                charm_path,
-                bases_index=pytestconfig.option.sysbench_charm_bases_index,
-            )
-        else:
-            return await _build_charm(charm_path)
-
-    ops_test.build_charm = build_charm
-    return ops_test
```
Comment (dragomirp) [inline tests/integration/conftest.py:31]: This is no longer needed by the reusable workflow and hides the extra params of the original `build_charm`

File: tests/integration/helpers.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "postgresql": {
         "charm": "postgresql",
         "channel": "14/edge",
-        "config": {},
+        "config": {"profile": "testing"},
```
Comment (dragomirp) [inline tests/integration/helpers.py:25]: Postgresql should also be using the testing profile

**Review by phvalguima — APPROVED** [review]

**Pull request merged**